x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java

```diff
@@ -686,4 +689,29 @@ public PlanFactory visitCompositeForkSubQuery(EsqlBaseParser.CompositeForkSubQue
         PlanFactory makePlan = typedParsing(this, ctx.forkSubQueryProcessingCommand(), PlanFactory.class);
         return input -> makePlan.apply(lowerPlan.apply(input));
     }
+
+    @Override
+    public PlanFactory visitRrfCommand(EsqlBaseParser.RrfCommandContext ctx) {
+        return input -> {
+            Source source = source(ctx);
+            Attribute scoreAttr = new UnresolvedAttribute(source, "_score");
+            Attribute forkAttr = new UnresolvedAttribute(source, "_fork");
+            Attribute idAttr = new UnresolvedAttribute(source, "_id");
+            Attribute indexAttr = new UnresolvedAttribute(source, "_index");
+            List<NamedExpression> aggregates = List.of(
```

Comment:
Could we avoid exposing the aggregations needed to the clients, and create the `Dedup` plan using only the needed attributes? It seems a bit confusing to me that the clients need to deal with aggregations and grouping - we could encapsulate that logic in `Dedup` itself.

Reply:
I kept it this way because the DEDUP step will likely become its own command.
What we do here would be the equivalent of doing something like:
`DEDUP ON _id, _index KEEP _score = SUM(_score)`.
We not only indicate that we want to dedup and which fields to use for that, but that we also want to replace the `_score` column with an aggregate value.
If we encapsulate everything in `Dedup`, we will likely need to expose the fields and aggregate options anyway at some point, taking out any custom logic that we made for RRF in `Dedup`. We will then need to go back to the `LogicalPlanBuilder` to change how RRF uses `Dedup`.

Reply:
I am tracking this as a follow up in the GH meta issue: https://github.com/elastic/elasticsearch/issues/123389 and some folks from the AE group are already aware that we have a dedup step.
My next step after this PR is to see how we can improve the dedup step since we are using `stats by` atm which is suboptimal and might not even be the behaviour we want.

Reply:
Thanks for clarifying Ioana!